File: chatwoot/csharp/src/OSEG.ChatwootExamples/CreateAWebhookExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Com.Chatwoot.Api;
using Com.Chatwoot.Client;
using Com.Chatwoot.Model;

namespace OSEG.ChatwootExamples;

public class CreateAWebhookExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey = new Dictionary<string, string> {["userApiKey"] = "USER_API_KEY"};
        // config.ApiKey = new Dictionary<string, string> {["agentBotApiKey"] = "AGENT_BOT_API_KEY"};
        // config.ApiKey = new Dictionary<string, string> {["platformAppApiKey"] = "PLATFORM_APP_API_KEY"};

        var webhookCreateUpdatePayload = new WebhookCreateUpdatePayload(
            url: null,
            subscriptions: [
            ]
        );

        try
        {
            var response = new WebhooksApi(config).CreateAWebhook(
                accountId: null,
                data: webhookCreateUpdatePayload
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling WebhooksApi#CreateAWebhook: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: chatwoot/typescript-axios/src/CreateAWebhookExample.ts
import axios, { AxiosError } from "axios";
import * as api from "chatwoot_client"

const configuration = new api.Configuration({
  apiKey: "USER_API_KEY",
  // apiKey: "AGENT_BOT_API_KEY",
  // apiKey: "PLATFORM_APP_API_KEY",
});

const webhookCreateUpdatePayload: api.WebhookCreateUpdatePayload = {
  subscriptions: [
  ],
};

new api.WebhooksApi(configuration).createAWebhook(
  0, // accountId
  webhookCreateUpdatePayload, // data
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling WebhooksApi#createAWebhook:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
